core/templates/dev/head/galleries/gallery.html

```diff
@@ -21,7 +21,12 @@
   <script type="text/javascript">
     GLOBALS.LANGUAGE_CODES_AND_NAMES = JSON.parse(
       '{{LANGUAGE_CODES_AND_NAMES|js_string}}');
-    GLOBALS.CAROUSEL_SLIDES_CONFIG = JSON.parse('{{CAROUSEL_SLIDES_CONFIG|js_string}}');
+    GLOBALS.CAROUSEL_SLIDES_CONFIG = JSON.parse(
+      '{{CAROUSEL_SLIDES_CONFIG|js_string}}');
+    {% if explorations_list %}
```

Comment:
I don't understand, why are we passing all this information through here? Usually we get it via JS.

In fact, I think maybe I see what's going on. There should be a GallerySearchPage, yes, but this page should just return a shell that then initiates an AJAX call to a GallerySearchHandler. The latter actually provides the page with the *data* to populate it with; this mimics what we are currently doing for the main gallery. I would suggest following this flow, since it's standard throughout the app (first we load an HTML page, which initiates an AJAX request to a JS handler).